### PR TITLE
Fix rte-france's API URL for GB parser

### PR DESCRIPTION
**Pull request by AlexandreCouderc**
## Description

Fix the GB's fetch_price parser by changing the API URL

### CODE REVIEW

**Comment by AlexandreCouderc** [comment]
Should fix https://github.com/electricitymaps/electricitymaps-contrib/issues/4556 (not tested)

**Comment by AlexandreCouderc** [comment]
`poetry run test_parser GB price` command's result is:

`2022-09-09 15:18:55,563 DEBUG    urllib3.connectionpool         Starting new HTTPS connection (1): www.rte-france.com:443
/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'www.rte-france.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
2022-09-09 15:18:55,634 DEBUG    urllib3.connectionpool         https://www.rte-france.com:443 "GET /themes/swi/xml/power-market-data.xml HTTP/1.1" 200 None
Parser result:
[{'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 0, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 270.49,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 1, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 248.11,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 2, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 238.41,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 3, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 212.45,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 4, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 206.33,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 5, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 230.8,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 6, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 251.57,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 7, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 368.99,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 8, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 413.13,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 9, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 400.55,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 10, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 363.39,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 11, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 328.66,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 12, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 323.12,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 13, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 297.5,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 14, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 289.3,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 15, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 306.73,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 16, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 318.04,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 17, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 389.47,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 18, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 428.25,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 19, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 450.06,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 20, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 450.75,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 21, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 413.7,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 22, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 380.82,
  'source': 'rte-france.com',
  'zoneKey': 'GB'},
 {'currency': 'EUR',
  'datetime': datetime.datetime(2022, 9, 9, 23, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Paris')),
  'price': 254.92,
  'source': 'rte-france.com',
  'zoneKey': 'GB'}]
---------------------
took 0.07s
min returned datetime: 2022-09-08T22:00:00+00:00 UTC
max returned datetime: 2022-09-09T21:00:00+00:00 UTC  -- OK, <2h from now :) (now=2022-09-09T13:18:55.636588+00:00 UTC)`

File: parsers/GB.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def fetch_price(
 
     url = ("https://www.rte-france.com/themes/swi/xml/power-market-data.xml")
 
-    response = r.get(url)
+    response = r.get(url, verify=False)
```
Comment (VIKTORVAV99) [inline parsers/GB.py:32]: Why are we not verifying SSL?

File: parsers/GB.py [hunk 2]
```diff
@@ -25,20 +25,11 @@ def fetch_price(
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> list:
-    if target_datetime:
```
Comment (VIKTORVAV99) [inline parsers/GB.py:28]: If we no longer can fetch historical data we should raise a not Implemented exception when a datetime is passed to the function.

**Review by VIKTORVAV99 — COMMENTED** [review]
It works but you have to run the formating suit on the parser (`poetry run format`) for the CI checks to pass.

I also have a few other notes/questions but thanks for fixing the parser!

File: parsers/GB.py (hunk 1 above)
Comment (AlexandreCouderc) [reply parsers/GB.py:32]: I get this stacktrace without setting this parameter:

```2022-09-09 15:51:20,432 DEBUG    urllib3.connectionpool         Starting new HTTPS connection (1): www.rte-france.com:443
Traceback (most recent call last):
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
    conn.connect()
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/connection.py", line 414, in connect
    self.sock = ssl_wrap_socket(
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/lib/python3.10/ssl.py", line 512, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.10/ssl.py", line 1070, in _create
    self.do_handshake()
  File "/usr/lib/python3.10/ssl.py", line 1341, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='www.rte-france.com', port=443): Max retries exceeded with url: /themes/swi/xml/power-market-data.xml (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/click/core.py", line 1134, in __call__
    return self.main(*args, **kwargs)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/click/core.py", line 1059, in main
    rv = self.invoke(ctx)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/click/core.py", line 1401, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/click/core.py", line 767, in invoke
    return __callback(*args, **kwargs)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/test_parser.py", line 60, in test_parser
    res = parser(*args, target_datetime=target_datetime, logger=getLogger(__name__))
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/parsers/lib/config.py", line 15, in wrapped_f
    result = f(*args, **kwargs)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/parsers/GB.py", line 32, in fetch_price
    response = r.get(url)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/requests/sessions.py", line 555, in get
    return self.request('GET', url, **kwargs)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/home/couderal/Perso/ElectricityMaps/electricitymap-contrib/.venv/lib/python3.10/site-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='www.rte-france.com', port=443): Max retries exceeded with url: /themes/swi/xml/power-market-data.xml (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))

File: parsers/GB.py (hunk 2 above)
Comment (AlexandreCouderc) [reply parsers/GB.py]: Your comment made me looking for how rte-france is fetching historical data, they use another URL. We only need to change this URL in the code.

File: parsers/GB.py (hunk 1 above)
Comment (VIKTORVAV99) [reply parsers/GB.py]: Strange, I checked in the browser and it seemed to have a valid SSL cert.

**Comment by AlexandreCouderc** [comment]
I don't understand why the formatting suit is still failing on your CI, it says no error on my local environment.

**Comment by VIKTORVAV99** [comment]
> I don't understand why the formatting suit is still failing on your CI, it says no error on my local environment.

Apparently it thinks flake8 is missing, which it shouldn't be...

Fortunately we run the new test in parallel with the old CI and the old CI passes so everything checks out. But @tonypls it seems like we have another CI issue, this looks like a problem with the cache/install.

I'll see if I can fix it in another PR.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Pull request merged**